cli/cmd/install.go

```diff
@@ -321,6 +322,11 @@ func render(config installConfig, w io.Writer, options *installOptions) error {
 	// Special case for linkerd-proxy running in the Prometheus pod.
 	injectOptions.proxyOutboundCapacity[config.PrometheusImage] = prometheusProxyOutboundCapacity
 
+	// Skip outbound port 443 to enable Kubernetes API access without the proxy.
+	// Once Kubernetes supports sidecar containers, this may be removed, as that
+	// will guarantee the proxy is running prior to control-plane startup.
+	injectOptions.ignoreOutboundPorts = []uint{443}
```

Comment:
Oooh, I forgot that you could also write this as:

```diff
diff --git a/cli/cmd/install.go b/cli/cmd/install.go
index 8b449544..e350f912 100644
--- a/cli/cmd/install.go
+++ b/cli/cmd/install.go
@@ -317,7 +317,7 @@ func render(config installConfig, w io.Writer, options *installOptions) error {
 	}
 
 	injectOptions := newInjectOptions()
-	injectOptions.proxyConfigOptions = options.proxyConfigOptions
+	*injectOptions.proxyConfigOptions = *options.proxyConfigOptions
 
 	// Special case for linkerd-proxy running in the Prometheus pod.
 	injectOptions.proxyOutboundCapacity[config.PrometheusImage] = prometheusProxyOutboundCapacity
```

Reply:
Nice catches. Updates pushed, testing now.

Related (not at all for this PR, mostly academic, and possibly controversial)... I think this class of issue could be somewhat addressed with more intentional use of pointers vs. structs. In this case, modifying `render` to take `options installOptions` would be clearer that `options` is not mutable. Unfortunately `installOptions` itself contains a `*proxyConfigOptions`. So ideally this could change to be structs all the way down, and we only pass them as pointers when we intend to mutate them.